ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/views/HiveMaterializedViewUtils.java

```diff
@@ -347,6 +355,38 @@ public static RelNode copyNodeNewCluster(RelOptCluster optCluster, RelNode node)
     }
   }
 
+  /**
+   * Validate if given materialized view has SELECT privileges for current user
+   * @param cachedMVTable
+   * @return false if user does not have privilege otherwise true
+   * @throws HiveException
+   */
+  public static boolean checkPrivilegeForMV(List<Table> cachedMVTableList) throws HiveException {
```

Comment:
Can we use full name: `checkPrivilegeForMaterializedViews` ?

Reply:
Done